sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggregationIterator.scala

```diff
@@ -157,31 +159,61 @@ abstract class AggregationIterator(
       inputAttributes: Seq[Attribute]): (InternalRow, InternalRow) => Unit = {
     val joinedRow = new JoinedRow
     if (expressions.nonEmpty) {
-      val mergeExpressions = functions.zip(expressions).flatMap {
-        case (ae: DeclarativeAggregate, expression) =>
-          expression.mode match {
+      val filterExpressions = expressions.map(_.filter)
+      val notExistsFilter = !filterExpressions.exists(_ != None)
+      var isFinalOrMerge = false
```

Comment:
Can you use `val` for `isFinalOrMerge` like this?
```
 val isFinalOrMerge = functions.exists(....)
```

Reply:
`isFinalOrMerge` is related to `expressions`.

Reply:
If you want to check if they have PartialMerge or Final;
```
      val isFinalOrMerge = expressions.map(_.mode)
        .collect { case PartialMerge | Final => true }.nonEmpty
```

Reply:
Then,  plz move this variable to line 223.

Reply:
good idea